tests/test_traits.py

```diff
@@ -324,3 +334,82 @@ class OptionalSymNameOp(IRDLOperation):
     assert interface.is_optional_symbol(symbol)
     symbol.verify()
     assert interface.get_sym_attr_name(symbol) == StringAttr("main")
+
+
+def test_symbol_table():
+    # Some helper classes
+    @irdl_op_definition
+    class SymbolTableOp(IRDLOperation):
+        name = "test.symbol_table"
+
+        sym_name = opt_attr_def(StringAttr)
+
+        one = region_def()
+        two = opt_region_def()
+
+        traits = frozenset([SymbolTable(), OptionalSymbolOpInterface()])
+
+    @irdl_op_definition
+    class SymbolOp(IRDLOperation):
+        name = "test.symbol"
+
+        sym_name = attr_def(StringAttr)
+
+        traits = frozenset([SymbolOpInterface()])
+
+    @irdl_op_definition
+    class TerminatorOp(IRDLOperation):
+        name = "test.terminator"
+
+        traits = frozenset([IsTerminator()])
```

Comment:
Can we reuse `test.termop` here?

Reply:
Thought dump, we probably should have a test.symbol and test.symbol_table too?

Reply:
We could, and for that there is something I would actually like to try.
So right now, MLIR puts all these methods in a `testd` dialect. But that means that the `testd` dialect is really big and it's really hard to know what's there.
On the other hand, what we could do is have a `test_symbol.py` file, that we could load dynamically (like having something like `xdsl-opt --load test_symbol.py my_file.mlir).
That could be useful to test these kind of things in filecheck.

But otherwise yes, we could put it in the `test` dialect in the meantime, if our goal is to put this in a filecheck test (otherwise, no one else will use it anyway so no need to).